src/utils/request.ts

```diff
@@ -1,41 +1,73 @@
 import respond from "../utils/respond";
 import errorMessages from "../constants/errorMessages"
 import { ImageKitOptions, UploadResponse } from "../interfaces";
+import IKResponse from "../interfaces/IKResponse";
 
 interface SignatureResponse {
     signature: string
     expire: number
     token: string
 }
 
-export const request = (formData: FormData, options: ImageKitOptions & { authenticationEndpoint: string }, callback?: (err: Error | null, response: UploadResponse | null) => void) => {
-    generateSignatureToken(options, (err, signaturObj) => {
+function getResponseHeaderMap(xhr: XMLHttpRequest) {
+    const headers: Record<string, string | number | boolean> = {};
+    xhr.getAllResponseHeaders()
+        .trim()
+        .split(/[\r\n]+/)
+        .map(value => value.split(/: /))
+        .forEach(keyValue => {
+          headers[keyValue[0].trim()] = keyValue[1].trim();
+        });
+    return headers;
+}
+
+const addResponseHeadersAndBody = (body: any, xhr: XMLHttpRequest):IKResponse<UploadResponse> => {
+    let response = { ...body };
+    const responseMetadata = {
+        statusCode: xhr.status,
+        headers: getResponseHeaderMap(xhr)
+    }
+    Object.defineProperty(response, "$ResponseMetadata", {
+        value: responseMetadata,
+        enumerable: false,
+        writable: false
+    });
+    return response as IKResponse<UploadResponse>;
+}
+
+export const request = (uploadFileXHR: XMLHttpRequest,formData: FormData, options: ImageKitOptions & { authenticationEndpoint: string }, callback?: (err: Error | null, response: UploadResponse | null) => void) => {
+    var signatureXHR = new XMLHttpRequest();
```

Comment:
Why do we need to create xhr for signature here and change `generateSignatureToken` at all?